src/figuresClasses.ts

```diff
@@ -1,19 +1,117 @@
-export interface Figure {
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
 
+enum Color {
+  Red = 'red',
+  Green = 'green',
+  Blue = 'blue',
 }
 
-export class Triangle {
+function roundArea(area: number): number {
+  return Math.floor(100 * area) / 100;
+}
+
+function isPositiveNumber(...numbers: number[]): boolean {
+  return numbers.some((num: number) => num <= 0);
+}
 
+export interface Figure {
+  color: Color;
+  shape: string;
+  getArea(): number;
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  public shape: Shape = Shape.Triangle;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (isPositiveNumber(a, b, c)) {
+      throw new Error('Sides are not valid (side <= 0)');
+    }
+
+    const sides = [this.a, this.b, this.c];
+
+    let longestSide: number = sides[0];
+
+    // eslint-disable-next-line no-plusplus
+    for (let i = 1; i < sides.length; i++) {
+      if (sides[i] > longestSide) {
+        longestSide = sides[i];
+      }
 
+      if (i === 2) {
+        sides.splice(sides.indexOf(longestSide), 1);
+      }
+    }
+
+    if (longestSide >= sides.reduce((
+      sum: number,
+      currentSide:number,
+    ) => sum + currentSide)) {
+      throw new Error(
+        'The longest side of a triangle is >= than a sum of two others',
+      );
+    }
+  }
```

Comment:
interesting solution, but better do it with one if and 3 conditions inside(3 lines of code).